modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/upgrade/v7_0_0/CommerceAddressUpgradeProcess.java

```diff
@@ -37,70 +36,103 @@ public class CommerceAddressUpgradeProcess extends UpgradeProcess {
 	public CommerceAddressUpgradeProcess(
 		AddressLocalService addressLocalService,
 		AccountEntryLocalService accountEntryLocalService,
+		CompanyLocalService companyLocalService,
 		ListTypeLocalService listTypeLocalService,
 		PhoneLocalService phoneLocalService,
 		UserLocalService userLocalService) {
 
 		_addressLocalService = addressLocalService;
 		_accountEntryLocalService = accountEntryLocalService;
+		_companyLocalService = companyLocalService;
 		_listTypeLocalService = listTypeLocalService;
 		_phoneLocalService = phoneLocalService;
 		_userLocalService = userLocalService;
 	}
 
 	@Override
 	protected void doUpgrade() throws Exception {
-		try (Statement selectStatement = connection.createStatement()) {
-			ResultSet resultSet = selectStatement.executeQuery(
-				"select * from CommerceAddress order by commerceAddressId");
-
-			while (resultSet.next()) {
+		_companyLocalService.forEachCompanyId(
+			companyId -> {
+				_setAddressListType(
+					companyId,
+					AccountListTypeConstants.
+						ACCOUNT_ENTRY_ADDRESS_TYPE_BILLING);
+				_setAddressListType(
+					companyId,
+					AccountListTypeConstants.
+						ACCOUNT_ENTRY_ADDRESS_TYPE_SHIPPING);
+				_setAddressListType(
+					companyId,
+					AccountListTypeConstants.
+						ACCOUNT_ENTRY_ADDRESS_TYPE_BILLING_AND_SHIPPING);
+			});
```

Comment:
We are adding the AddressListTypes in advance to avoid duplicate entry exceptions.